l1-contracts/src/core/Rollup.sol

```diff
@@ -658,27 +681,81 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
     for (uint256 i = 0; i < feesLength; i++) {
       publicInputs[9 + i] = _fees[i];
     }
-    uint256 feesEnd = 9 + feesLength;
+    uint256 offset = 9 + feesLength;
 
     // vk_tree_root
-    publicInputs[feesEnd] = vkTreeRoot;
+    publicInputs[offset] = vkTreeRoot;
+    offset += 1;
 
     // protocol_contract_tree_root
-    publicInputs[feesEnd + 1] = protocolContractTreeRoot;
+    publicInputs[offset] = protocolContractTreeRoot;
+    offset += 1;
 
     // prover_id: id of current epoch's prover
-    publicInputs[feesEnd + 2] = _args[6];
+    publicInputs[offset] = _args[6];
+    offset += 1;
+
+    // blob_public_inputs
+    uint256 blobOffset = 0;
+    for (uint256 i = 0; i < _epochSize; i++) {
+      uint8 blobsInBlock = uint8(_blobPublicInputsAndAggregationObject[blobOffset++]);
+      // asserting here to avoid looping twice in one fn
+      {
+        // Blob public inputs are 112 bytes long - see _validateBlobs() for explanation
+        bytes32 calcBlobPublicInputsHash = sha256(
+          abi.encodePacked(
+            _blobPublicInputsAndAggregationObject[blobOffset:blobOffset + 112 * blobsInBlock]
+          )
+        );
+        require(
+          calcBlobPublicInputsHash == blobPublicInputsHashes[previousBlockNumber + i + 1],
+          Errors.Rollup__InvalidBlobPublicInputsHash(
+            blobPublicInputsHashes[previousBlockNumber + i + 1], calcBlobPublicInputsHash
+          )
+        );
+      }
+      for (uint256 j = 0; j < Constants.BLOBS_PER_BLOCK; j++) {
+        if (j < blobsInBlock) {
+          // z
+          publicInputs[offset++] =
+            bytes32(_blobPublicInputsAndAggregationObject[blobOffset:blobOffset += 32]);
+          // y
+          (publicInputs[offset++], publicInputs[offset++], publicInputs[offset++]) =
+          _bytes32ToBigNum(
+            bytes32(_blobPublicInputsAndAggregationObject[blobOffset:blobOffset += 32])
+          );
+          // To fit into 2 fields, the commitment is split into 31 and 17 byte numbers
+          // TODO: The below left pads, possibly inefficiently
```

Comment:
Is there an issue for this? You don't need to make the work, but if you can add an issue and throw it into #7820 that would be neat 👍

Reply:
Made one, thanks! https://github.com/AztecProtocol/aztec-packages/issues/10148